Implemented the SR-IOV Multi VF device pass-through for MultiVM's.

santwana opened the pull request:
Implemented the SR-IOV Multi VF device pass-through for MultiVM's.
User can now create the desired no.of VM's along with the specified VF's(single or multiple),
to be pass-throughe'd to single/mutiple VM's.

Signed-off-by: Santwana Samantray <santwana@linux.vnet.ibm.com>

santwana commented:
[job.log](https://github.com/autotest/tp-libvirt/files/2256041/job.log)

balamuruhans commented on libvirt/tests/src/libvirt_pci_passthrough_multi.py:
```diff
@@ -0,0 +1,170 @@
+            i = 0
+            for vm in vms:
+                for _ in range(int(no_vf)):
+                    pci_xml = NodedevXML.new_from_dumpxml(vf1[i])
```
```python
for each in range(int(no_vf)):
    pci_xml = NodedevXML.new_from_dumpxml(vf1[each])
```

balamuruhans commented on libvirt/tests/src/libvirt_pci_passthrough_multi.py:
```diff
@@ -0,0 +1,170 @@
+            output = session.cmd_output("lspci -nn")
+            nic_list_after = output.splitlines()
+            net_ip = netaddr.IPAddress(net_ip)
+            if nic_list_after == nic_list_before:
```
It is better to sort the list for asserting,

balamuruhans commented on libvirt/tests/src/libvirt_pci_passthrough_multi.py:
```diff
@@ -0,0 +1,170 @@
+            if nic_list_after == nic_list_before:
+                test.fail("passthrough Adapter not found in guest.")
+            else:
+                logging.debug("Adapter passthorughed to guest successfully")
```
typo for passthrough

balamuruhans commented on libvirt/tests/src/libvirt_pci_passthrough_multi.py:
```diff
@@ -0,0 +1,170 @@
+                    "test in config file")
+    for vm in vms:
+        vmxml = VMXML.new_from_inactive_dumpxml(vm.name)
+        backup_xml = vmxml.copy()
```
backup_xml for each vm should be saved in a dict, else each iteration will overwrite backup_xml

balamuruhans commented on libvirt/tests/src/libvirt_pci_passthrough_multi.py:
```diff
@@ -0,0 +1,170 @@
+            pci_xml = NodedevXML.new_from_dumpxml(val)
+            pci_address = pci_xml.cap.get_address_dict()
+            vmxml.add_hostdev(pci_address)
+            vmxml.sync()
```
In multivm env, to which vmxml we are adding this hostdev ?

balamuruhans commented on libvirt/tests/src/libvirt_pci_passthrough_multi.py:
```diff
@@ -0,0 +1,170 @@
+                net_ip = net_ip + 1
+
+    finally:
+        backup_xml.sync()
```
sync() the xml for all the vms

santwana commented:
depends on: autotest/tp-libvirt#2179

santwana commented:
@balamuruhans I have addressed the review changes and also rebased the base file to existing (https://github.com/autotest/tp-libvirt/pull/2179/). Kindly review. Thanks.

sathnaga commented:
@santwana needs rebase.

dzhengfy commented:
After confirmation with the IBM PPC FVT lead that the author does not work on this any more and the pr is not in use by them. So I close it.
